pandas/core/internals/array_manager.py

```diff
@@ -482,15 +521,19 @@ def is_single_block(self) -> bool:
 
     def get_bool_data(self, copy: bool = False) -> ArrayManager:
         """
-        Parameters
-        ----------
-        copy : bool, default False
-            Whether to copy the blocks
+        Select columns that are bool-dtype and object-dtype columns that are all-bool.
         """
-        mask = np.array([is_bool_dtype(t) for t in self.get_dtypes()], dtype="object")
-        arrays = [self.arrays[i] for i in np.nonzero(mask)[0]]
+        arrays = []
+        indices = []
+        for i, arr in enumerate(self.arrays):
```

Comment:
why is this written in a completely different way than get_numeric_data? (in fact *both*) of these should just use a helper that you can pass a predicate function (IIRC that's how we do it in BM, though maybe mis-remembering)

Reply:
Just because I rewrote here only `get_bool_data`. I don't think BlockManager is currently doing what you say, but that's a good idea for here as a precursor -> https://github.com/pandas-dev/pandas/pull/40030